### PR TITLE
meta: Add file based META streams

### PR DESCRIPTION
**Pull request by lmars**
This makes META streams first class citizens in go-meta and managed by the store.

META streams are new-line separated lists of CIDs stored in a local file, writers append to the end of the file and readers tail the file waiting for new CIDs to be written.

There is a stream HTTP API at `/stream` which supports reading and writing to a META stream. Reads are a stream of Server-Sent-Events, with each event being the object payload, and writes are a POST request to append to the stream:

* write to the "foo.meta" stream:

```
$ curl -X POST -H "Content-Type: application/json" --data-binary '{"value": 1}' http://localhost:5000/stream/foo.meta
{"/":"zdpuAwhdyitq3qhsdhmCepsiWmXHbSnVWSevyuzjtLx6gpZGB"}

$ curl -X POST -H "Content-Type: application/json" --data-binary '{"value": 2}' http://localhost:5000/stream/foo.meta
{"/":"zdpuB2kFRbN7wCw71XetiwTQdJ8xL1sWTPtprapDu5M2vvA9P"}

$ curl -X POST -H "Content-Type: application/json" --data-binary '{"value": 3}' http://localhost:5000/stream/foo.meta
{"/":"zdpuB19ATtxLsSMi2FR8AKbY1CUR3iYDY1PiEuucYmD6Qs8Vn"}
```

* read the stream:

```
$ curl http://localhost:5000/stream/foo.meta
data: {"value":1}

data: {"value":2}

data: {"value":3}
```

### CODE REVIEW

File: stream/client.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"errors"
+	"net/http"
+
+	"github.com/flynn/flynn/pkg/httpclient"
+	"github.com/flynn/flynn/pkg/stream"
+	"github.com/ipfs/go-cid"
+)
+
+type Client struct {
```
Comment (orenyodfat) [inline stream/client.go]: Comment public types and functions.

File: stream/client.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+func NewClient(baseURL string) *Client {
+	return &Client{&httpclient.Client{
+		HTTP:        http.DefaultClient,
+		URL:         baseURL,
+		ErrNotFound: errors.New("stream: not found"),
+	}}
+}
+
+func (c *Client) ReadStream(name string, outputCh interface{}) (stream.Stream, error) {
```
Comment (orenyodfat) [inline stream/client.go]: Comment public types and functions.

File: stream/client.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	return c.Stream("GET", "/"+name, nil, outputCh)
+}
+
+func (c *Client) WriteStream(name string, v interface{}) (*cid.Cid, error) {
```
Comment (orenyodfat) [inline stream/client.go]: Comment public types and functions.

**Comment by MrTibbles** [comment]
Can the README be updated for mere mortals, such as i?

**Comment by lmars** [comment]
@orenyodfat comments added in 86f3db6.

@MrTibbles I will overhaul the README after finishing the onslaught of changes :smile:.

**Pull request merged**